app/src/main/java/a3/com/convo/activities/LoginActivity.java

```diff
@@ -189,22 +198,36 @@ public void done(ParseException e) {
     }
 
     protected void getFriendsOnApp(AccessToken access_token) {
-        // TODO: add the friends' object IDs not facebook ids
         GraphRequest request = GraphRequest.newMyFriendsRequest(
                 access_token,
                 new GraphRequest.GraphJSONArrayCallback() {
                     @Override
                     public void onCompleted(JSONArray friends, GraphResponse response) {
                         try {
-                            ParseUser user = ParseUser.getCurrentUser();
+                            final ParseUser user = ParseUser.getCurrentUser();
                             // initialize empty likes array
                             user.put("friends", new ArrayList<String>());
                             for (int i = 0; i < friends.length(); i++) {
                                 JSONObject friend = friends.optJSONObject(i);
                                 String name = friend.optString("name");
-                                String id = friend.optString("id");
-                                user.add("friends", id);
-                                user.saveInBackground();
+                                final String id = friend.optString("id");
+                                ParseQuery<ParseUser> query = ParseUser.getQuery();
+                                query.whereEqualTo("username", id);
```

Comment:
Make the string "username" as static constant in the class so we can reuse this string.